### PR TITLE
feat(post): mocking soap services using nock

### PR DESCRIPTION
**Pull request by deeptiagrawa**
Cover page:
image

Article:
image

### CODE REVIEW

File: _posts/2018-10-02-mocking-soap-apis-in-functional-tests-using-nock.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+functional tests became unreliable.
+
+Here are the reasons why:
+1. The dependent services were slow.
+1. The dependent services were flaky.
+1. Covering corner cases with real test data is challenging. An example: Your
+   service has a special handling for data returned by external service which
+   contains products having last day of their free trial.
+
+Using real data became more of a pain for our CICD process. It also made our
```
Comment (silasbw) [inline _posts/2018-10-02-mocking-soap-apis-in-functional-tests-using-nock.md]: "real data" -> "live APIs"?

File: _posts/2018-10-02-mocking-soap-apis-in-functional-tests-using-nock.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+Nock is an HTTP server mocking library for Node.js applications.
+
+Nock intercepts HTTP request made using Node's `http.request` or
+`http.ClientRequest` modules and responds with the expected/mocked status code
+and response.
+
+## Nock setup
+
+Using Nock is simple. Run below command on terminal.
```
Comment (silasbw) [inline _posts/2018-10-02-mocking-soap-apis-in-functional-tests-using-nock.md]: You're explaining how to install Nock here, not use it.

"Install Nock by running:"

File: _posts/2018-10-02-mocking-soap-apis-in-functional-tests-using-nock.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+
+```sh
+npm i --save-dev nock
+```
+
+## Mocking REST service
+
+Considering the service under development has an endpoint `/user` which returns
```
Comment (silasbw) [inline _posts/2018-10-02-mocking-soap-apis-in-functional-tests-using-nock.md]: Considering -> Assume?

File: _posts/2018-10-02-mocking-soap-apis-in-functional-tests-using-nock.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+    }
+  })
+    .post('/HolidayService_v2/HolidayService2.asmx')
+    .times(2)
+    .replyWithFile(200, path.join(__dirname, './mocks/GetCountriesAvailable.xml'));
+```
+
+## Conclusion
+
+Using nock will make your functional test less flaky and more independent of
```
Comment (silasbw) [inline _posts/2018-10-02-mocking-soap-apis-in-functional-tests-using-nock.md]: You capitalize Nock elsewhere.
